src/wsiToDcm.cpp

```diff
@@ -177,18 +180,65 @@ int WsiToDcm::dicomizeTiff(
     uint32_t row = 1;
     uint32_t column = 1;
     int32_t levelToGet = level;
-    double downsample = 1.;
+    int64_t downsample = 1;
     if (retile) {
       if (downsamples.size() > level && downsamples[level] >= 1) {
-        downsample = downsamples[level];
+        downsample = static_cast<int64_t>(downsamples[level]);
       } else {
-        downsample = ((int64_t)pow(2, level));
+        downsample = static_cast<int64_t>(pow(2, level));
```

Comment:
We should probably just change the original flag to be an int, if we're going to cast it to an int later anyway. Fine to leave to a follow up PR so we can get this one in.